### ISSUE TITLE
Neural Networks From Scratch - victorzhou.com

### ISSUE THREAD

**Issue by guevara** [guevara/read-it-later#8638]
Neural Networks From Scratch - victorzhou.com<br>
<br>
https://ift.tt/KoIG1AH<br>
<br>
Victor Zhou<br>
<br>
<div>
<div>
<div>
<div>
<div>
<p><time>July 30, 2019</time><span>&ensp;|&ensp;UPDATED <time>February 9, 2020</time></span></p>
</div>
<div>
<p><img src="https://victorzhou.com/media/nn-series/network.svg"></p>
<p>This 4-post series, written especially with beginners in mind, provides a <strong>fundamentals-oriented</strong> approach towards understanding Neural Networks. We&rsquo;ll start with an introduction to <strong>classic Neural Networks</strong> for complete beginners before delving into two popular variants: <strong>Recurrent Neural Networks</strong> (RNNs) and <strong>Convolutional Neural Networks</strong> (CNNs).</p>
<p>For each of each these types of networks, we&rsquo;ll:</p>
<ul>
<li>See the <strong>structure</strong> of the network.</li>
<li>Understand the <strong>motivation</strong> behind using that type of network.</li>
<li>Introduce a <strong>real-world problem</strong> that can be solved using that network.</li>
<li>
<strong>Manually derive the gradients</strong> needed to train our problem-specific network.</li>
<li>
<strong>Implement a fully-functioning network completely from scratch</strong> (using only <a rel="noopener noreferrer" href="https://numpy.org/" target="_blank">numpy</a>) in Python.</li>
</ul>
<h2>
<a href="https://victorzhou.com/series/neural-networks-from-scratch/#background"></a>Background</h2>
<p><strong>This series requires ZERO prior knowledge of Machine Learning</strong> or Neural Networks. However, background in the following topics may be helpful:</p>
<ul>
<li>
<strong>Multivariable Calculus</strong>, used when deriving the gradients needed to train our networks. These gradient derivations can be skipped if you don&rsquo;t have the background.</li>
<li>
<strong>Linear Algebra</strong>, specifically Matrix algebra - matrices are often the best way to represent weights for Neural Networks.</li>
<li>
<strong>Python 3</strong>, because the Python implementations in these posts are a major part of their educational value. A baseline proficiency in Python is enough.</li>
</ul>
<h2>
<a href="https://victorzhou.com/series/neural-networks-from-scratch/#the-series"></a>The Series</h2>
<p>Ready to get started? Here we go:</p>
</div>
</div>
</div>
</div>
</div>
<br>
<br>
<br>
<br>
<br>
<br>
via victorzhou.com<br>
<br>
August 26, 2022 at 06:27PM
